src/mtev_websocket_client.h

```diff
@@ -24,49 +24,153 @@ typedef struct {
+                                  const char *path, const char *service,
+                                  mtev_websocket_client_callbacks *callbacks,
+                                  void *closure,
+                                  eventer_pool_t *pool,
+                                  mtev_hash_table *sslconfig);
+
+/*! \fn void mtev_websocket_client_set_ready_callback(mtev_websocket_client_t *client, mtev_websocket_client_ready_callback ready_callback)
+    \brief Set a new ready_callback on an existing client
+    \param client client to modify
+    \param ready_callback new ready_callback to set
+*/
 API_EXPORT(void)
   mtev_websocket_client_set_ready_callback(mtev_websocket_client_t *client,
-                                           mtev_websocket_client_ready_callback msg_callback);
+                                           mtev_websocket_client_ready_callback ready_callback);
 
+/*! \fn void mtev_websocket_client_set_msg_callback(mtev_websocket_client_t *client, mtev_websocket_client_msg_callback msg_callback)
+    \brief Set a new msg_callback on an existing client
+    \param client client to modify
+    \param msg_callback new msg_callback to set
+*/
 API_EXPORT(void)
   mtev_websocket_client_set_msg_callback(mtev_websocket_client_t *client,
                                          mtev_websocket_client_msg_callback msg_callback);
 
+/*! \fn void mtev_websocket_client_set_cleanup_callback(mtev_websocket_client_t *client, mtev_websocket_client_cleanup_callback cleanup_callback)
+    \brief Set a new cleanup_callback on an existing client
+    \param client client to modify
+    \param cleanup_callback new cleanup_callback to set
+*/
 API_EXPORT(void)
   mtev_websocket_client_set_cleanup_callback(mtev_websocket_client_t *client,
                                              mtev_websocket_client_cleanup_callback cleanup_callback);
 
+/*! \fn mtev_boolean mtev_websocket_client_send(mtev_websocket_client_t *client, int opcode, void *buf, size_t len)
+    \brief Enqueue a message
+    \param client
+    \param opcode opcode as defined in RFC 6455 and referenced in wslay.h
+    \param buf pointer to buffer containing data to send
+    \param len number of bytes of buf to send
+    \return boolean indicating success/failure
+
+    This function makes a copy of buf of length len.
+    This function may fail for the following reasons:
+    1. The client was not ready. See mtev_websocket_client_is_ready.
+    2. The client was already closed. See mtev_websocket_client_is_closed.
+    3. Out of memory.
+*/
 API_EXPORT(mtev_boolean)
   mtev_websocket_client_send(mtev_websocket_client_t *client, int opcode,
                              void *buf, size_t len);
 
+/*! \fn void mtev_websocket_client_free(mtev_websocket_client_t *client)
+    \brief Free a client
+    \param client client to be freed
+    This function will cleanup the client(and hence trigger any set cleanup_callback) first.
```

Comment:
Sorry to be unclear the first time.  The block of description must be separated from the  slash commands (or the text will be wrapped up into the last command).